static/js/search.js

```diff
@@ -1161,6 +1161,31 @@ function populateResults(data, full, jumpToSingle) {
       div.textContent = `Number of results: ${count} (maximum is around 4000)`;
       header.append(div);
     }
+
+    // Detect "definitions exist but no uses"
+    let hasDefinitions = false;
+    let hasUses = false;
+
+    for (const [_, counts] of tupledCounts.entries()) {
+      if (_.includes("Definition") && counts.files > 0) {
+        hasDefinitions = true;
+      }
+      if (_.includes("Use") && counts.hits > 0) {
+        hasUses = true;
+      }
+    }
+
+    if (hasDefinitions && !hasUses) {
+      const div = document.createElement("div");
+      div.classList.add("search-note");
+      div.innerHTML = `
+        <strong>No uses found.</strong>
+        <div class="search-note-muted">
+          Note: Some template-heavy or indirect usages may not be fully indexed.
```

Comment:
There's slight mismatch between the class names.

The top-level box says "search-note", that sounds like this box is a note,
and there are two boxes, one with `strong`, and one with
"muted" class name with "Note:" prefix.

In terms of the text content, the "search-note-muted" box sounds like the main content of this box, given it has "Note:" prefix, and in that case the "strong" part sounds more like a title or something.

Maybe we could rename the top-level box "search-warning" or something, that way the <strong> part looks more suitable as a main content, and then rename `search-note-muted` to `search-warning-note`, so that that matches the content with "Note:" prefix.